### ISSUE TITLE
Create user problem

### ISSUE THREAD

**Issue by fadiabualnaser** [JuanDMeGon/RESTful-API-with-Laravel-Definitive-Guide#1]
I try to create new user but i can't because the store action doesn't receive the password attribute so i check the Transforminput middleware and fount that the first foreach need to fix at accept the attributes if the attribute not in originalAttribute like password and password_confirmation.

my fix is 

        foreach ($request->request->all() as $input => $value) {
            if( $transformer::originalAttribute($input) ){
                $transformedInput[$transformer::originalAttribute($input)] = $value;
            }
            else{
                $transformedInput[$input] = $value;
            }
        }

**Comment by JuanDMeGon** [comment]
Hello @fadiabualnaser,
Thanks for open an issue about this, I was thinking about doing the same for the different problems reported on the course.

Based on your solution, this may allow sending whatever attribute, which may include the original values from the model and not from the transformer.

For example, if the user/client send "name" instead of "title" this will pass as it is and will be accepted but should not. I mean, if the user/client send title it should pass, but if it sends name, it should not pass because if it passes this will allow knowing the original attributes from the database which is not intended.

I think that a better approach is just to include the password and passwprd_conformation fields on the transformer for User.

Thanks again, and let me know your thoughts.

Best wishes.

**Comment by fadiabualnaser** [comment]
I was thinking the same after i wrote this issue, I agree with you it's better to add to needed attributes to transformer.

**Comment by JuanDMeGon** [comment]
Yeah, do not worry about that. Thanks for opening the issue, in that way I'll be able to track the different problems and, in fact, the students can comment and propose solutions.
Best wishes.

**Comment by JuanDMeGon** [comment]
Hello, 
Include password and password_confirmation in the arrays of the transformation should work. 
Try to check the questions on the course to find further details.
Best wishes.
